### PR TITLE
Feature Request : Introduction of An DependsDataSrc Endpoint

### PR DESCRIPTION
**Pull request by mohanajuhi166**
Feature Request Part one for the following

/data-sources/{ds_id}/execution-plan-ids -- a collection of the execution plan IDs directly associated with the given data source.

Feature Issue : https://github.com/AbsaOSS/spline/issues/793

### CODE REVIEW

File: producer-services/src/main/scala/za/co/absa/spline/producer/service/repo/ExecutionProducerRepository.scala [hunk 1]
```diff
@@ -24,4 +24,5 @@ trait ExecutionProducerRepository {
   def insertExecutionPlan(executionPlan: ExecutionPlan)(implicit ec: ExecutionContext): Future[Unit]
   def insertExecutionEvents(executionEvents: Array[ExecutionEvent])(implicit ec: ExecutionContext): Future[Unit]
   def isDatabaseOk()(implicit ec: ExecutionContext): Future[Boolean]
+  def getDataSources(value: String, access: Optional[String])(implicit ec: ExecutionContext): Future[Array[String]]
```
Comment (cerveada) [inline producer-services/src/main/scala/za/co/absa/spline/producer/service/repo/ExecutionProducerRepository.scala]:
```suggestion
  def getDataSources(value: String, access: Option[String])(implicit ec: ExecutionContext): Future[Array[String]]
```
- Optional is Java class
- Option is Scala class

**Comment by wajda** [comment]
I think it should be a part of Consumer API, not Producer API. Could you please move it?

File: producer-services/src/main/scala/za/co/absa/spline/producer/service/repo/ExecutionProducerRepository.scala (hunk 1 above)
Comment (mohanajuhi166) [reply producer-services/src/main/scala/za/co/absa/spline/producer/service/repo/ExecutionProducerRepository.scala]: making the change

**Comment by mohanajuhi166** [comment]
Adding the changes as per requested

**Pull request closed without merging**